app/indexers/dataset_indexer.rb

```diff
@@ -13,7 +13,12 @@ class DatasetIndexer < Hyrax::WorkIndexer
   # Enable queries to distinguish top-level vs. nested datasets
   def generate_solr_document
    super.tap do |solr_doc|
-     solr_doc['top_level_bsi'] = object.in_works_ids.any? ? false : true
+     if object.in_works_ids.any?
+       solr_doc['top_level_bsi'] = false
+       solr_doc[Rdr::Index::Fields.in_works_ids] = object.in_works_ids
+     else
+       solr_doc['top_level_bsi'] = true
+     end
```

Comment:
For the `if` block, consider:

```ruby
solr_doc['top_level_bsi'] = object.in_works_ids.blank? # or empty?
solr_doc[Rdr::Index::Fields.in_works_ids] = object.in_works_ids
```

Reply:
@dchandekstark I do like this better.  PR revised accordingly.  (It did require a slight change in a spec test.)